src/components/AddToMetamask/index.tsx

```diff
@@ -2,27 +2,38 @@ import React from 'react'
 import { Token } from '@trisolaris/sdk'
 import styled from 'styled-components'
 
-import { StyledMenuButton } from '../../components/StyledMenu'
+import { StyledMenuButton } from '../StyledMenu'
 import { MouseoverTooltip } from '../Tooltip'
+import { LinkStyledButton } from '../../theme'
 
 import { useActiveWeb3React } from '../../hooks'
 
 import { registerToken } from '../../utils/wallet'
 
-import MetamaskIcon from '../../assets/images/metamask.png'
+import MetaMaskIcon from '../../assets/images/metamask.png'
 
-const StyledMetamaskButton = styled(StyledMenuButton)`
+const StyledMetaMaskButton = styled(StyledMenuButton)`
   width: 35px;
 `
-const StyledMetamaskImg = styled.img`
- display:flex;
+
+const StyledMetaMaskImg = styled.img`
+  display: flex;
   width: 20px;
   :hover {
     cursor: pointer;
   }
 `
 
-function AddToMetamaskButton({ token, noBackground, ...otherProps }: { token: Token; noBackground?: boolean }) {
+function AddToMetaMaskButton({
+  token,
+  noBackground,
+  textOnly,
+  ...otherProps
+}: {
+  token: Token
+  noBackground?: boolean
+  textOnly?: boolean
+}) {
   const { library } = useActiveWeb3React()
 
   if (!token) {
```

Comment:
Isn't `token` a required prop?

Reply:
Indeed, I added it as a guard to avoid errors in runtime in case the trade token isn't loaded yet. The other option would be handlng it directly on Swap, like

```
currencies.OUTPUT && <AddToMetamask token={currencies.OUTPUT}/>
```